### ISSUE TITLE
How to use built-in optimizers for PerAvg

### ISSUE THREAD

**Issue by Jiamian-Wang** [TsingZ0/PFL-Non-IID#76]
Hi there,

Thanks for the great work! 

I'm currently focusing on the PerFedAvg and noticed that the Optimizer is customized as PerAvgOptimizer(). 
It seems that the step1 and 2 in the train() in the clientperavg.py exactly implement the PerAvg core updating functionality. 
Are there any specific concerns to implement a PerAvgOptimizer()? If so, could you please suggest what are the potential challenges to use the default optimizer like Adam()? 

Thank you so much for the kind response!

**Comment by TsingZ0** [comment]
I write code based on two goals: easy-to-understand and baseline identity. Except for the core method design, we keep other settings, such as optimizer, learning rate, etc., the same as FedAvg. 

This platform is easy to extend. You can create a new `PerAvgOptimizer` and use Adam for your own experiments.
